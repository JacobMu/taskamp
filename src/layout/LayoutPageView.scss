.layoutPage {
  display: flex;
  flex-direction: initial;
}

.childFlexGrow {
  flex-grow: 8;
}
