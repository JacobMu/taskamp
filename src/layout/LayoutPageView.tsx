import React  from "react";

import { SideMenuBarContainer } from "./side-menu/SideMenuBarContainer";
import "./LayoutPageView.scss";
import { ButtonProps } from "./side-menu/SideMenuService";
import { TaskProps } from "./dialog/CreateNewTaskDialogService";

interface Props {
  children: React.ReactNode;
  buttons: ButtonProps[];
  dispatchNewCard(cardProps: TaskProps): void;
}

export const LayoutPageView: React.FC<Props> = ({
  children,
  buttons,
  dispatchNewCard,
}) => {
  return (
    <div className="layoutPage">
      <SideMenuBarContainer
        buttons={buttons}
        dispatchNewCard={dispatchNewCard}
      />
      <div className="childFlexGrow">{children}</div>
    </div>
  );
};
