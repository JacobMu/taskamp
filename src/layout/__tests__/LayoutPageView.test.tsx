import React from "react";
import { shallow } from "enzyme";
import { LayoutPageView } from "../LayoutPageView";

const DEFAULT_PROPS = {
  buttons: [],
  dispatchNewCard: jest.fn().mockName("dispatchNewCardMock"),
};

describe("<LayoutPageView />", () => {
  it(" renders layout page", () => {
    const wrapper = shallow(
      <LayoutPageView {...DEFAULT_PROPS}>
        <div>Children Mock</div>
      </LayoutPageView>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
