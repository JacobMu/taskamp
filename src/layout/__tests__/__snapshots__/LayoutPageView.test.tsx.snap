// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutPageView />  renders layout page 1`] = `
<div
  className="layoutPage"
>
  <SideMenuBarContainer
    buttons={Array []}
    dispatchNewCard={[MockFunction dispatchNewCardMock]}
  />
  <div
    className="childFlexGrow"
  >
    <div>
      Children Mock
    </div>
  </div>
</div>
`;
